src/meta/service/src/ddl_service.rs

```diff
@@ -827,3 +840,18 @@ fn fill_table_source(
     table.optional_associated_source_id =
         Some(OptionalAssociatedSourceId::AssociatedSourceId(source_id));
 }
+
+// `server.id` (in the range from 1 to 2^32 - 1). This value MUST be unique across whole replication
+// group (that is, different from any other server id being used by any master or slave)
+fn fill_cdc_mysql_server_id(fragment_graph: &mut PbStreamFragmentGraph) {
+    let rand_server_id = rand::thread_rng().gen_range(1..4294967295u32);
+    for fragment in fragment_graph.fragments.values_mut() {
+        visit_fragment(fragment, |node_body| {
+            if let NodeBody::Source(source_node) = node_body {
+                let props = &mut source_node.source_inner.as_mut().unwrap().properties;
+                props.insert("server.id".to_string(), rand_server_id.to_string());
```

Comment:
Will this overwrite the user-specified server.id if they set it already?

Reply:
After consideration, I think we should still allow user to set the ID by their own, so I will revert here back to generate a server.id when it is not provided. 
Btw I think it doesn't make sense to check the uniqueness of server.id in the validation, since we cannot **lock** the server.id to avoid other connector to use it at the time we create a connector. IMO we should provide some ways to improve observability for this run time errors.